pkg/controller/deployment/util/deployment_util.go

```diff
@@ -554,7 +554,7 @@ func FindOldReplicaSets(deployment *extensions.Deployment, rsList []*extensions.
 
 // WaitForReplicaSetUpdated polls the replica set until it is updated.
 func WaitForReplicaSetUpdated(c clientset.Interface, desiredGeneration int64, namespace, name string) error {
-	return wait.Poll(10*time.Millisecond, 1*time.Minute, func() (bool, error) {
+	return wait.PollImmediate(10*time.Millisecond, 1*time.Minute, func() (bool, error) {
```

Comment:
Yeouch!  Do we have an issue to change this to either a targeted watch a poll on the cache?